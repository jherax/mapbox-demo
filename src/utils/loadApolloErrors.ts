import {loadDevMessages, loadErrorMessages} from '@apollo/client/dev';

export default function loadApolloErrors() {
  if (!['prod', 'production'].includes(process.env.NODE_ENV)) {
    loadDevMessages();
    loadErrorMessages();
  }
}
