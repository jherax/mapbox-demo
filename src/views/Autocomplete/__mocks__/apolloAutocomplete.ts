import {ApolloError} from '@apollo/client';
import {MockedResponse} from '@apollo/client/testing';
import {GraphQLError} from 'graphql';

import {CITIES_BY_COUNTRY} from '../services/getCitiesByCountry';
import citiesMock from './cities.json';

export const variablesMock = {country: 'Colombia'};

/**
 * @@delay: ms to prevent React from batching the loading state away.
 * If you only want to test the loading state, set value to Infinity.
 * @see https://www.apollographql.com/docs/react/development-testing/testing
 */
export const successCitiesByCountryMock: MockedResponse[] = [
  {
    request: {
      query: CITIES_BY_COUNTRY.query,
      // ...CITIES_BY_COUNTRY.options,
      variables: variablesMock,
    },
    result: {
      data: {
        getCitiesByCountry: {
          message: 'Mocked results',
          result: citiesMock,
        },
      },
    },
  },
];

export const emptyCitiesByCountryMock: MockedResponse[] = [
  {
    request: {
      query: CITIES_BY_COUNTRY.query,
      variables: variablesMock,
    },
    result: {
      data: {
        getCitiesByCountry: {
          message: 'Mocked results',
          result: [],
        },
      },
    },
  },
];

export const errorCitiesByCountryMock: MockedResponse[] = [
  {
    request: {
      query: CITIES_BY_COUNTRY.query,
      variables: variablesMock,
    },
    error: new ApolloError({
      graphQLErrors: [
        new GraphQLError('Unknown server error when fetching data'),
      ],
    }),
  },
];
