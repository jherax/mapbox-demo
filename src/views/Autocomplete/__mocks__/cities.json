[
  {
    "name": "Sant Julià de Lòria",
    "lat": "42.46372",
    "lng": "1.49129",
    "country": "CO",
    "admin1": "06",
    "admin2": ""
  },
  {
    "name": "Pas de la Casa",
    "lat": "42.54277",
    "lng": "1.73361",
    "country": "CO",
    "admin1": "03",
    "admin2": ""
  },
  {
    "name": "Ordino",
    "lat": "42.55623",
    "lng": "1.53319",
    "country": "CO",
    "admin1": "05",
    "admin2": ""
  },
  {
    "name": "Les Escaldes",
    "lat": "42.50729",
    "lng": "1.53414",
    "country": "CO",
    "admin1": "08",
    "admin2": ""
  },
  {
    "name": "la Massana",
    "lat": "42.54499",
    "lng": "1.51483",
    "country": "CO",
    "admin1": "04",
    "admin2": ""
  },
  {
    "name": "Encamp",
    "lat": "42.53474",
    "lng": "1.58014",
    "country": "CO",
    "admin1": "03",
    "admin2": ""
  },
  {
    "name": "Canillo",
    "lat": "42.5676",
    "lng": "1.59756",
    "country": "CO",
    "admin1": "02",
    "admin2": ""
  },
  {
    "name": "Arinsal",
    "lat": "42.57205",
    "lng": "1.48453",
    "country": "CO",
    "admin1": "04",
    "admin2": ""
  },
  {
    "name": "Andorra la Vella",
    "lat": "42.50779",
    "lng": "1.52109",
    "country": "CO",
    "admin1": "07",
    "admin2": ""
  }
]
