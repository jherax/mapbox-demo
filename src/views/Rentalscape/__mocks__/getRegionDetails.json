{
  "region": {
    "items": [
      {
        "regionName": "ca-placer",
        "name": "Placer County, CA",
        "logo": "https://str-placer-county-ca-deploy-us-west-2-prod.s3-us-west-2.amazonaws.com/PlacerCALogo.png",
        "misc": "{}",
        "geoJson": "https://public-portal-assets-prod-usw2.s3.us-west-2.amazonaws.com/shapefiles/ca-placer.geojson",
        "legends": [
          {
            "colorHex": "#00AAAB",
            "count": 3334,
            "formattedText": "Licensed Short-Term Rental properties",
            "tooltip": null,
            "propertyType": "LICENSED"
          }
        ],
        "statistics": [],
        "links": [
          {
            "displayValue": "530-448-8003",
            "label": "24/7 hotline: ",
            "tooltip": null,
            "type": "PHONE",
            "value": "530-448-8003"
          },
          {
            "displayValue": "strcompliance@placer.ca.gov",
            "label": "Complaint email: ",
            "tooltip": null,
            "type": "EMAIL",
            "value": "strcompliance@placer.ca.gov"
          }
        ]
      }
    ]
  }
}
