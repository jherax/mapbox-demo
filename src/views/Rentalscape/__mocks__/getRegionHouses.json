{
  "region": {
    "properties": {
      "nextToken": "eyJ2ZXJzaW9uIjozLCJ0b2tlbiI6IkFnVjRVQXB0V0d5TEp2cFZCcVZhZkVoMFZRVzJiQjlUcGJLN1NaMEFMZFc4VVU0QWV3QUNBQWRCY0hCVGVXNWpBQkZGYm1OeWVYQjBhVzl1UTI5dWRHVjRkQUFWWVhkekxXTnllWEIwYnkxd2RXSnNhV010YTJWNUFFUkJhazlsVEVOb1RtMXdNMjl2YVc1TGJrUkRNRU4yVkZkbGRHRlhWVEZJTjJ0SGJIZHZVUzlGT1hoVmVIWlVWM1pLUW5nelNXWmlla2xPYzAxWWFYZG1Va0U5UFFBQkFBZGhkM010YTIxekFFdGhjbTQ2WVhkek9tdHRjenAxY3kxM1pYTjBMVEk2T1RBeU1URTNNelEwTXpNek9tdGxlUzh4TmpVek1EaGlOaTB3WTJNNExUUmpOMkV0WW1FMFlTMHhaRGswWXpNNU9EZzJPRFVBdUFFQ0FRQjRlZUdaL2RxRUV6WEtsQk1rNWtWQWFXYTFvSTdHdyt4M251UXhQTlZ4RjVFQmdWQUxGVE1EelB1ZE5RSVZSRCtTQlFBQUFINHdmQVlKS29aSWh2Y05BUWNHb0c4d2JRSUJBREJvQmdrcWhraUc5dzBCQndFd0hnWUpZSVpJQVdVREJBRXVNQkVFREZXZ0dXb2pVRGxaSW5zbnJ3SUJFSUE3NlJuQmhWOWtmc2NhUFF1VU5FelllQ1lKODEzcmJ0ZkEreWN5VXNPMS9PTmt4MUZ6L25FUitMbDdnaDJGdEx0aUJwRXppbXQra2ZtTUwwZ0NBQUFRQUtvR05XdnlTaWpJaWlkL2ZBeDdsQVNydWJQcDBYZ2JQVXBTVWJ5QlVpVFpLZXFHaXJMMzc4SWFFbG9qYnlWTEVmLy8vLzhBQUFBQkFBQUFBQUFBQUFBQUFBQUJBQUFDUmltQ2NwbmpxTGd0aldtZjBuUjlyWEJIZFRSOVdRYnZHVFBGaTN1TEVoY0lhRUNxVnBPZWQ3cmRxa1RnVDloQmZCSDBqbmZYQTA0V2tSZG1ETE04Z2ZyN29CcnMxUmhZUlQ1U0ozT3hJcmxZa1B2TTR1RTRHOWhBMjJnejY0Nk1CMTBBcldoUVhna1dIU1FLSVp5THBmSkFLem5MeFFvVVBCWk1aUGY2OGlwcHRDV0UzYkU1SG9obXoxSDJCYnlsSEs2OFlMdmtQSkM2UFFxYVlnTGxKM3J2N0pXcjhkVlpwbk9ZZ1pya1JYTUtwQzkvS3pUaXYvZCtHRU5rWUgxeWs4NnF5a3MyQUVhSjZZbDNxNGFSRVFWL1paVSsyR280bUhsd3FrRzBEUzVtUDhoTDlwUmhGQ1dORDBPVXluK25Fd3dxVXJNRHB2ampWZFE4a2dDNzhHa3QraEZaUDMycWM5QmliWWtvSDl4WmpiTlc2UHkzOUFEeG5LU1Brb0tqd210VkZidTJhT29OUWFSSCtIaklxcnNCZTJQMTJFWmthOEFxRXdhRGova3hSM2hFeitMOVo3THhIMm9rYWk5MlMxRks2NWRyWERtV05iYUpaN1VUTnMvbllwZkxKUkRnWkVYd1NJN3RwbWkzTW5ZUU80SklCY3k5SDZ5TXpLemRvc2x0dmE3dExUbzhxNWNXQlRVRkJiZHBzYWNCaHdxdzc3VTZRU3ZXWStnc2dEWVRTbWhZdDhGWTBFcWd2RkdhT2pkODBseCtheGkrWEkwb2wrejgxZW85bDBON0Qydnd6V2Vpa3lKaENxQThwZFA1YU5QN2JnNjgrNFEwRDdvSllOS2RNeVNYRmVlOXVIbTQxNHdsYmJBRlBVYlRxa2JNU3UzNWN6NWRtczF2R2kzUjVuMWlrdnpQdk95KzFkTWFGUXc2ZmdLRWhodFpKWUY4bzRtbXo5SHFtcFpQSE1rNTZqNzg1WFNPcitFd1FZVlNkOFl6UWl6ZEVUa3l0YSt6cFJRSVk1Q0hKWlBqbzd4MEkxUzV4SS9xclFvaXloVUFaekJsQWpCUGZLQTAvNVZRWDBmdDBOZTBWSlN6ZmdwYXlkOEpSU0xRU0dKS2owcktmQTdLVGtoTjlQT1lKSkVsM051bm9lOENNUUM2TEhTdUlBU1E2Vm9PTHF0MG0xOU1zTmlDUU1OWk9GdHNCSkpHeWFIVXBucUpBcSt6bDBhUUdlU3JiUDBCWHM4PSJ9",
      "items": [
        {
          "deckardId": "APN-CA-PLACER-093400014000",
          "displayInfo": [
            {
              "label": "License",
              "value": "STR22-5918"
            }
          ],
          "isLicensed": true,
          "types": ["LICENSED"],
          "geo": {
            "lat": 39.190257458488425,
            "lng": -120.11636275650973
          },
          "geo5": "GEO5STRING",
          "address": "2755 N Lake Blvd 36 Tahoe City CA 96145",
          "contact": null,
          "licenses": {
            "items": []
          },
          "listings": {
            "items": []
          }
        },
        {
          "deckardId": "APN-CA-PLACER-093400035000",
          "displayInfo": [
            {
              "label": "License",
              "value": "STR22-6857"
            }
          ],
          "isLicensed": true,
          "types": ["LICENSED"],
          "geo": {
            "lat": 39.19000062771026,
            "lng": -120.11588242993494
          },
          "geo5": "GEO5STRING",
          "address": "2755 N Lake Blvd 11 Tahoe City CA 96145",
          "contact": null,
          "licenses": {
            "items": []
          },
          "listings": {
            "items": []
          }
        },
        {
          "deckardId": "APN-CA-PLACER-095390021000",
          "displayInfo": [
            {
              "label": "License",
              "value": "STR22-6092"
            }
          ],
          "isLicensed": true,
          "types": ["LICENSED"],
          "geo": {
            "lat": 39.17125422794592,
            "lng": -120.23263719897322
          },
          "geo5": "GEO5STRING",
          "address": "2101 Scott Peak Pl 10 Alpine Meadows CA 96146",
          "contact": null,
          "licenses": {
            "items": []
          },
          "listings": {
            "items": []
          }
        }
      ]
    }
  }
}
