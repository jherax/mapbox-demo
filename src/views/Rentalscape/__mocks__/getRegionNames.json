{
  "listRegions": {
    "items": [
      {
        "name": "City of South Padre Island, TX",
        "regionShortName": "townofsouthpadreisland"
      },
      {
        "name": "Placer County, CA",
        "regionShortName": "placer-ca"
      },
      {
        "name": "Cathedral City, CA",
        "regionShortName": "ca-cathedralcity"
      },
      {
        "name": "City of Palm Desert, CA",
        "regionShortName": "cityofpalmdesert"
      }
    ]
  }
}
