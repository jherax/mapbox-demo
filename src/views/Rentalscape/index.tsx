import styled from 'styled-components';

import Mapbox from '../../components/Mapbox';
import Sidebar, {type SidebarProps} from '../../components/Sidebar';
import logger from '../../utils/logger';
import setHrefValue from '../../utils/setHrefValue';
import trim from '../../utils/trim';
import ResolveView from '../ResolveView';
import {useRegionDetails, useRegionHouses} from './services/hooks';

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: start;
  align-items: stretch;
  height: 100vh;
  width: 100%;
  box-sizing: border-box;
`;

function Rentalscape() {
  const regionDetails = useRegionDetails();
  const regionHouses = useRegionHouses();

  logger.info({
    details: {loading: regionDetails.loading, data: regionDetails.data},
    houses: {loading: regionHouses.loading, data: regionHouses.data},
  });

  const firstItem = regionDetails.data?.region?.items[0];
  const sidebarProps: SidebarProps = {
    regionLogo: firstItem?.logo ?? '',
    legends: (firstItem?.legends ?? []).map(item => {
      return {
        count: item.count,
        label: trim(item.formattedText),
        bulletColor: item.colorHex,
      };
    }),
    links: (firstItem?.links ?? []).map(item => {
      return {
        label: trim(item.label),
        showValue: trim(item.displayValue),
        rawValue: setHrefValue({type: item.type, value: item.value}),
      };
    }),
  };

  return (
    <Wrapper>
      <ResolveView loading={regionDetails.loading} error={regionDetails.error}>
        <Sidebar {...sidebarProps}></Sidebar>
        <Mapbox></Mapbox>
      </ResolveView>
    </Wrapper>
  );
}

export default Rentalscape;
