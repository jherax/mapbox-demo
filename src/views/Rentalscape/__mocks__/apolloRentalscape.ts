import {ApolloError} from '@apollo/client';
import {MockedResponse} from '@apollo/client/testing';
import {GraphQLError} from 'graphql';

import {REGION_DETAILS} from '../services/getRegionDetails';
import {REGION_HOUSES} from '../services/getRegionHouses';
import regionDetailsMock from './getRegionDetails.json';
import regionHousesMock from './getRegionHouses.json';

/**
 * @@delay: ms to prevent React from batching the loading state away.
 * If you only want to test the loading state, set value to Infinity.
 * @see https://www.apollographql.com/docs/react/development-testing/testing
 */
export const successRegionDetailsMock: MockedResponse[] = [
  {
    delay: 30,
    request: {
      query: REGION_DETAILS.query,
      ...REGION_DETAILS.options,
    },
    result: {
      data: regionDetailsMock,
    },
  },
];

export const emptyRegionDetailsMock: MockedResponse[] = [
  {
    delay: 30,
    request: {
      query: REGION_DETAILS.query,
      ...REGION_DETAILS.options,
    },
    result: {
      data: {region: {items: []}},
    },
  },
];

export const errorRegionDetailsMock: MockedResponse[] = [
  {
    request: {
      query: REGION_DETAILS.query,
      ...REGION_DETAILS.options,
    },
    error: new ApolloError({
      graphQLErrors: [
        new GraphQLError('Unknown server error when fetching data'),
      ],
    }),
  },
];

// ---------------------------------

export const successRegionHousesMock: MockedResponse[] = [
  {
    delay: 30,
    request: {
      query: REGION_HOUSES.query,
      ...REGION_HOUSES.options,
    },
    result: {
      data: regionHousesMock,
    },
  },
];

export const emptyRegionHousesMock: MockedResponse[] = [
  {
    delay: 30,
    request: {
      query: REGION_HOUSES.query,
      ...REGION_HOUSES.options,
    },
    result: {
      data: {region: null},
    },
  },
];

export const errorRegionHousesMock: MockedResponse[] = [
  {
    request: {
      query: REGION_HOUSES.query,
      ...REGION_HOUSES.options,
    },
    error: new ApolloError({
      graphQLErrors: [
        new GraphQLError('Unknown server error when fetching data'),
      ],
    }),
  },
];
