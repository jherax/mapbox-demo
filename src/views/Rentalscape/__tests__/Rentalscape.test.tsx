import {MockedProvider} from '@apollo/client/testing';
import {
  cleanup,
  queryByAttribute,
  render,
  screen,
} from '@testing-library/react';

import loadApolloErrors from '../../../utils/loadApolloErrors';
import logger from '../../../utils/logger';
import Rentalscape from '..';
import * as dataMocks from '../__mocks__/apolloRentalscape';

jest.mock('react-map-gl');
jest.spyOn(logger, 'info');

loadApolloErrors();
const getById = queryByAttribute.bind(null, 'id');

/**
 * @see https://jestjs.io/docs/snapshot-testing
 * @see https://testing-library.com/docs/react-testing-library/setup
 */
describe('Testing <Rentalscape />', () => {
  afterEach(() => cleanup());

  it('should render correctly when there are no fetched data', async () => {
    const emptyGqlMocks = [
      dataMocks.emptyRegionDetailsMock,
      dataMocks.emptyRegionHousesMock,
    ].flat();

    const {asFragment} = render(
      <MockedProvider mocks={emptyGqlMocks} addTypename={false}>
        <Rentalscape />
      </MockedProvider>,
    );

    expect(await screen.findByText('Loading...')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  // ---------------------------------

  it('should render Sidebar and Mapbox sections', async () => {
    const graphqlMocks = [
      dataMocks.successRegionDetailsMock,
      dataMocks.successRegionHousesMock,
    ].flat();

    const {asFragment, container} = render(
      <MockedProvider mocks={graphqlMocks} addTypename={false}>
        <Rentalscape />
      </MockedProvider>,
    );

    expect(await screen.findByText('Loading...')).toBeInTheDocument();
    expect(await screen.findByText('PUBLIC PORTAL')).toBeInTheDocument();
    expect(getById(container, 'Sidebar')).toBeInTheDocument();
    expect(getById(container, 'Mapbox')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  // ---------------------------------

  it('should render the error view', async () => {
    const errorGqlMocks = [
      dataMocks.errorRegionDetailsMock,
      dataMocks.errorRegionHousesMock,
    ].flat();

    const {asFragment} = render(
      <MockedProvider mocks={errorGqlMocks} addTypename={false}>
        <Rentalscape />
      </MockedProvider>,
    );

    const errorMessage = 'Unknown server error when fetching data';
    expect(await screen.findByText('Loading...')).toBeInTheDocument();
    expect(await screen.findByText(errorMessage)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
