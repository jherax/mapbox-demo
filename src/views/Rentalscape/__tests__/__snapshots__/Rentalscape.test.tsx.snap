// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Rentalscape /> should render Sidebar and Mapbox sections 1`] = `
<DocumentFragment>
  <div
    class="sc-fsYfdN jhFlPF"
  >
    <section
      class="sc-guDLey bJAjUB"
      id="Sidebar"
    >
      <header
        class="sc-dmyCSP gVCNvA"
      >
        <section
          id="box-titles"
        >
          <div
            id="box-image"
          >
            <img
              alt="Rentalscape"
              src="https://placer-ca-str-public-portal.deckard.com/rental-scape-logo.png"
            />
          </div>
          <div
            id="box-title"
          >
            <h1>
              Short-Term Rental (STR)
            </h1>
            <h2>
              PUBLIC PORTAL
            </h2>
          </div>
        </section>
        <section
          id="box-region-logo"
        >
          <img
            alt="region logo"
            src="https://str-placer-county-ca-deploy-us-west-2-prod.s3-us-west-2.amazonaws.com/PlacerCALogo.png"
          />
        </section>
      </header>
      <article
        class="sc-hLQSwg VgcZZ"
      >
        <section>
          <h2>
            Program Stats
          </h2>
          <ul>
            <li
              class="sc-eDLKkx eglPZG"
            >
              <div>
                <span />
                <label />
              </div>
            </li>
          </ul>
        </section>
      </article>
      <article
        class="sc-jTQCzO kOGkHE"
      >
        <section>
          <h2>
            Useful Links
          </h2>
          <ul>
            <li>
              <span />
              <a
                href=""
                rel="noopener noreferrer nofollow"
              />
            </li>
            <li>
              <span />
              <a
                href=""
                rel="noopener noreferrer nofollow"
              />
            </li>
          </ul>
        </section>
      </article>
      <footer
        class="sc-gLLuof dTnuzm"
      >
        <p>
          Powered by Rentalscape
        </p>
        <p>
          Engineered by DECKARD TECHNOLOGIES
        </p>
      </footer>
    </section>
    <section
      class="sc-beySPh jhvjtQ"
      id="Mapbox"
    />
  </div>
</DocumentFragment>
`;

exports[`Testing <Rentalscape /> should render correctly when there are no fetched data 1`] = `
<DocumentFragment>
  <div
    class="sc-fsYfdN jhFlPF"
  >
    <section
      class="sc-iBdnpw dByDQe ResolveView_loading"
    >
      <pre>
        Loading...
      </pre>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Testing <Rentalscape /> should render the error view 1`] = `
<DocumentFragment>
  <div
    class="sc-fsYfdN jhFlPF"
  >
    <section
      class="sc-iBdnpw dByDQe ResolveView_error"
    >
      <pre>
        Unknown server error when fetching data
      </pre>
    </section>
  </div>
</DocumentFragment>
`;
